### DISCUSSION TITLE
Facet filtering with negative conditions

### DISCUSSION THREAD

**Discussion by Kerollmops** [meilisearch/product#111] [Feedback & Feature Proposal]
### Is your feature request related to a problem? Please describe.

Facet filtering would be really useful when you want the engine to ignore certain documents.

[For example you want to mark documents as read by given users](https://github.com/meilisearch/MeiliSearch/discussions/832), you therefore want to maintain a list of users ids in each of the documents. This users ids can be used to skip these documents, they represent the users ids who already read the documents.

### Describe the solution you'd like

There is two ways of fixing this, usings filters and using facet filtering:
 - the filter syntax could be improved to allow filtering on arrays (e.g. `NOT (read_by_users CONTAINS my_id)`).
 - the facet filters could be used with negative conditions (e.g. `["-read_by_users:my_id"]`).

The second option would be way more performant than the first solution I described.

**Comment by gmourier** [comment]
Locking this old discussion.

📚 https://docs.meilisearch.com/learn/advanced/filtering_and_faceted_search.html#filter-expressions
